pkg/build/generator/generator.go

```diff
@@ -261,6 +261,16 @@ func (g *BuildGenerator) Instantiate(ctx kapi.Context, request *buildapi.BuildRe
 	}
 	glog.V(4).Infof("Build %s/%s has been generated from %s/%s BuildConfig", newBuild.Namespace, newBuild.ObjectMeta.Name, bc.Namespace, bc.ObjectMeta.Name)
 
+	// Add the BuildConfig to the Build's OwnerReference, so that it will be
+	// deleted when the BuildConfig is deleted
+	// TODO: is there a utility function for generating an owner reference from an object?
+	newBuild.OwnerReferences = append(newBuild.OwnerReferences, kapi.OwnerReference{
+		APIVersion: "v1",          // BuildConfig.APIVersion is not populated
```

Comment:
we need to sort out the right way to resolve this.... we've got an internal buildconfig object, we (apparently?) need to know the api version of the actual buildconfig, or maybe just the api version of the server we're running in, to set this correctly?

@deads2k help
@jim-minter you've got a similar block in your template PR